.github/workflows/beam_PreCommit_Go.yml

```diff
@@ -1,37 +1,57 @@
-name: beam_PreCommit_Go
+name: Go Precommit
 
 on:
   push:
     tags: ['v*']
     branches: ['master', 'release-*']
-  pull_request:
-    branches: ['master']
+  pull_request_target:
+    branches: ['master', 'release-*' ]
     paths: ['model/**', 'sdks/go.**', 'release/**']
   issue_comment:
     types: [created]
   schedule:
     - cron: '* */6 * * *'
+permissions:
+  actions: write
+  pull-requests: read
+
 
 jobs:
   beam_PreCommit_Go:
-        if: ${{github.event.issue.pull_request}} || ${{github.event.comment.body == 'Run Go PreCommit'}} || ${{github.event.schedule}}
-        runs-on: [self-hosted, ubuntu-20.04]
-        name: beam_PreCommit_Go
-        steps:       
-          - name: Git checkout
-            uses: actions/checkout@v3
-          - name: Install Java
-            uses: actions/setup-java@v3.8.0
-            with:
-              distribution: 'zulu'
-              java-version: '8'
-          - name: Install Go
-            uses: actions/setup-go@v4
-            with:
-              go-version: '1.20'
-          - name: Setup Gradle
-            uses: gradle/gradle-build-action@v2
-            with:
-              cache-read-only: false
-          - name: run goPreCommit script
-            run: ./gradlew :goPreCommit
\ No newline at end of file
+    if: |
+      github.event_name == 'push' ||
+      github.event_name == 'pull_request_target' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run Go PreCommit'
+    runs-on: [self-hosted, ubuntu-20.04]
+    name: beam_PreCommit_Go
+    steps:       
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Install GH CLI
+        uses: dev-hanz-ops/install-gh-cli-action@v0.1.0
+      - name: Rerun on comment
+        if: github.event.comment.body == 'Run Go PreCommit'
+        uses: ./.github/actions/rerun-job-action
+        with:
+          pull_request_url: ${{ github.event.issue.pull_request.url }}
+          github_repository: ${{ github.repository }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ github.job }}
+          github_current_run_id: ${{ github.run_id }}
+      - name: Install Java
+        uses: actions/setup-java@v3.8.0
+        with:
+          distribution: 'zulu'
+          java-version: '8'
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
```

Comment:
Can we use the `Setup self-hosted` step instead of `Install Java` and `Install Go` steps?

- name: Setup self-hosted
   uses: ./.github/actions/setup-self-hosted-action
   with:
     requires-py-38: false
     requires-py-39: false

Reply:
Plan is to soon start using a prepared image which will have dependencies already installed so i dont think there is much value now for this